packages/core/src/internal/fetchOptions.ts

```diff
@@ -75,6 +76,15 @@ export const makeFetchURL = (
   return finalUrl;
 };
 
+/** Extract the query parameters from a URL, without using {@link URL} to support partial URLs */
+const extractParams = (url: string): [string, URLSearchParams] => {
+  const start = url.indexOf('?');
+  if (start === -1) return [url, new URLSearchParams()];
+
+  const raw = url.slice(start + 1);
+  return [url.slice(0, start), new URLSearchParams(raw)];
```

Comment:
```suggestion
  return start > -1
    ? [url.slice(0, start), new URLSearchParams(url.slice(start + 1))]
    : [url, new URLSearchParams()]
```

nit: this is just a little different than our typical code style. Nothing major, but since I've got more suggestions we can align this with the rest